gammapy/irf/psf/map.py

```diff
@@ -16,6 +17,25 @@
 __all__ = ["PSFMap", "RecoPSFMap"]
 
 
+PSF_MAX_OVERSAMPLING = 4
+
+
+def _psf_upsampling_factor(psf, geom, position, energy=None, precision_factor=12):
+    if energy is None:
+        energy = geom.axes[psf.energy_name].center
+    psf_r68 = psf.containment_radius(
+        0.68, geom.axes[psf.energy_name].center, position=position
+    )
+    psf_r68 = np.percentile(psf_r68, 50)
+    base_factor = (2 * psf_r68 / geom.pixel_scales.max()).to_value("")
+    factor = np.minimum(
+        int(np.ceil(precision_factor / base_factor)), PSF_MAX_OVERSAMPLING
+    )
+    if isinstance(geom, HpxGeom):
```

Comment:
Maybe an "if else" can be introduced, in order to avoid to compute psf.containment_radius in case of HEALPix

Reply:
here the if statement makes the computed factor a multiple of 2 so it can work in Healpix but we still need to compute the factor as in the wcs case